utam-compiler/src/main/java/utam/compiler/grammar/UtamElement.java

```diff
@@ -213,7 +209,11 @@ private UtamComposeMethod buildWaitForComposeMethod() {
             null);
     List<UtamMethodAction> compose =
         Collections.singletonList(new UtamMethodActionWaitForElement(name, isLoad()));
-    return new UtamComposeMethod(methodName, description, compose, !isPrivateWait());
+
+    // q: why does this determine the public nature?
+    Boolean isPublic = !isLoad() || isWait();
+
+    return new UtamComposeMethod(methodName, description, compose, isPublic);
```

Comment:
isWait() instead isPublic

Reply:
Going to leave this one. I think it's important for the reader to easily see that isWait is being mapped to isPublic.